### ISSUE TITLE
Allow developer to define custom label for Pagination's pageSizes list

### ISSUE THREAD

**Issue by akmalmzamri** [carbon-design-system/carbon#6387]
### Summary

Currently, in Pagination component, we can provide a list of page size to the prop `propSizes`

```
<Pagination
    totalItems={totalItems}
    backwardText="Previous page"
    forwardText="Next page"
    page={currentPage}
    pageSize={currentPageSize}
    pageSizes=[5, 10, 15, 25]
    itemsPerPageText="Items per page" />

```

However, there's no way to define a custom text to the value of the pageSizes. In my case, I want to implement a `All` selection to display all data. I can manually handle `All` selection to display all data. However, the problem is with the other labels in the Pagination components which became NaN due to the dropdown having 'All' value instead of the value of the `totalItems`.

![image](https://user-images.githubusercontent.com/10990690/86550034-d9890000-bf73-11ea-8bd2-54cea72b1973.png)

### Justification

One of the use case is to allow user to make selection to all data at once (e.g. for Export, Delete)

### Desired UX and success metrics

Basically, developer should be able to define the dropdown value and text separately, e.g:

```
const pageSizeList = [
    {value: 5, text: '5'},
    {value: 10, text: '10'},
    {value: 15, text: '15'},
    {value: 25, text: '25'},
    {value: totalItems, text: 'All'},
]
```
